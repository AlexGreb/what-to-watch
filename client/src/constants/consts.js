export const tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const buttonTypes = {
  SUBMIT: `submit`,
  RESET: `reset`,
  BUTTON: `button`
};

export const VIDEO_PROGRESS = 100;

export const MOVIES_TO_SHOW = 8;

export const BASE_URL = `https://15.react.htmlacademy.pro/wtw/`;

// export const BASE_URL = `http://localhost:3002/api/v1/`;

export const API_TIMEOUT = 5000;

export const Errors = {
  UNAUTHORIZED: 401
};
//  TODO брать из сервера
export const dataUrl = {
  HOME: `/`,
  FILMS: `/films/`,
  PROMO: `/promo`,
  LOGIN: `/login`,
  FAVORITE: `/favorite/`,
  PLAYER: `/player/`,
  REVIEWS: `/comments/`,
  // FILMS: `/movies`,
  // PROMO: `/promo`,
  // LOGIN: `/login`,
  // FAVORITE: `/favorite/`,
  // PLAYER: `/player/`,
  // REVIEWS: `/comments/`,
};


export const authorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};
