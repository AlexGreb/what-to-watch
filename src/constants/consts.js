export const tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const MOVIES_TO_SHOW = 8;
