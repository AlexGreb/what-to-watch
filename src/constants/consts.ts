export enum Tabs {
  OVERVIEW = `Overview`,
  DETAILS =`Details`,
  REVIEWS = `Reviews`
};

export enum ButtonTypes {
  SUBMIT = `submit`,
  RESET = `reset`,
  BUTTON = `button`
};

export const VIDEO_PROGRESS:number = 100;

export const MOVIES_TO_SHOW:number = 8;

export const BASE_URL:string = `https://5.react.pages.academy/wtw/`;

export const API_TIMEOUT:number = 5000;

export enum Errors {
  UNAUTHORIZED = 401
};

export enum AppRoute {
  HOME = `/`,
  FILMS = `/films/`,
  PROMO = `/films/promo`,
  LOGIN = `/login`,
  FAVORITE = `/favorite/`,
  PLAYER = `/player/`,
  REVIEWS = `/comments/`,
};

export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`
};
