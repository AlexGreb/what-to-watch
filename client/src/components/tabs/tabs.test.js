/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render} from '@testing-library/react';
import Tabs from './Tabs.jsx';
import {Tabs} from '../../constants/consts.ts';

describe(`Tabs`, () => {
  it(`Should render correctly Tabs`, () => {
    const tabDetails = () => <div></div>;
    const {asFragment} = render(<Tabs tabNavs={Object.values(Tabs)}
      onChangeActiveTab={jest.fn()}
      tabDetails={tabDetails()}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});

