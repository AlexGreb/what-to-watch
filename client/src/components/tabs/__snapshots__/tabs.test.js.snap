// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Should render correctly Tabs 1`] = `
<DocumentFragment>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div />
</DocumentFragment>
`;
